### ISSUE TITLE
How to input a Double?

### ISSUE THREAD

**Issue by oxinabox** [google-research/dex-lang#497]
Literals seem to be `Float`.
as far as i can tell the [parser.dx code](https://github.com/google-research/dex-lang/blob/a373663ce4fb4b88d7f304f0bb2849eb518bb4c9/lib/parser.dx) doesn't support `Double` (or `Float`)

**Comment by danieldjohnson** [comment]
I believe you've linked to the prototype of pure-Dex userspace parsing, which isn't actually used when parsing normal Dex code. But indeed the Haskell parser implementation that parses Dex code seems to hard-code all non-integer numeric literals as floats:

https://github.com/google-research/dex-lang/blob/a373663ce4fb4b88d7f304f0bb2849eb518bb4c9/src/lib/Parser.hs#L209

(as does the type inference pass:)

https://github.com/google-research/dex-lang/blob/a373663ce4fb4b88d7f304f0bb2849eb518bb4c9/src/lib/Inference.hs#L295

**Comment by oxinabox** [comment]
> I believe you've linked to the prototype of pure-Dex userspace parsing, which isn't actually used when parsing normal Dex code.

Yes, as a work around to not being able to input them directly I thought I might use `parse Double "3.1415926535"`.

> But indeed the Haskell parser implementation that parses Dex code seems to hard-code all non-integer numeric literals as floats:

Yes that's the main issue
